packages/exoflex/src/types.ts

```diff
@@ -47,59 +47,62 @@ export type Theme = {
     textinput: boolean;
   };
   style?: {
-    accordion: Pick<
+    accordion?: Pick<
       AccordionProps<Title>,
       | 'containerStyle'
       | 'iconStyle'
       | 'sectionContainerStyle'
       | 'titleContainerStyle'
       | 'titleStyle'
     >;
-    activityIndicator: Pick<ActivityIndicatorProps, 'style'>;
-    avatarIcon: Pick<AvatarIconProps, 'style'>;
-    avatarImage: Pick<AvatarImageProps, 'style'>;
-    avatarText: Pick<AvatarTextProps, 'style' | 'labelStyle'>;
-    badge: Pick<BadgeProps, 'style'>;
-    button: Pick<ButtonProps, 'contentStyle' | 'labelStyle' | 'style'>;
-    calendar: Pick<CalendarProps, 'style'>;
-    checkbox: Pick<CheckboxProps, 'style' | 'textStyle'>;
-    chip: Pick<ChipProps, 'style' | 'iconStyle' | 'textStyle'>;
-    collapsible: Pick<
+    activityIndicator?: Pick<ActivityIndicatorProps, 'style'>;
+    avatarIcon?: Pick<AvatarIconProps, 'style'>;
+    avatarImage?: Pick<AvatarImageProps, 'style'>;
+    avatarText?: Pick<AvatarTextProps, 'style' | 'labelStyle'>;
+    badge?: Pick<BadgeProps, 'style'>;
+    button?: Pick<ButtonProps, 'contentStyle' | 'labelStyle' | 'style'> & {
+      fontWeight?: FontWeight;
+      fontStyle?: FontStyle;
```

Comment:
Hmm, do you think we need to tell the user that this will not work if children is not a string?

Also, we also have `labelStyle` here, which is essentially the same thing but nested inside. Maybe we should pick one to use?

Reply:
> Hmm, do you think we need to tell the user that this will not work if children is not a string?

I'm planning to add that to the docs.

> Also, we also have labelStyle in this theme.style.button, which is essentially the same thing. Maybe we should pick one to use?

Changing either `fontWeight` or `fontStyle` through the `labelStyle` won't work as expected as it will not change to the correct `fontFamily`.

That's why I added both `fontWeight` and `fontStyle` so we can put it on the `Text` props to be used by `resolveTextStyle()`.

Reply:
Actually, we can also spread them from `labelStyle` using `StyleSheet.flatten`. From UX perspective, I think it's better but I'm kinda unsure to use `StyleSheet.flatten` only for that matter.

Reply:
Hmm, I think we don't need to flatten the style from theme.

I'm convinced with your approach.

I have one more idea, we can make the `theme.text.style` to be `TextStyle` instead of `StyleProp<TextStyle>` (no array etc), so now we can do this inside `Text`:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/19742419/81915691-23e9a280-95fd-11ea-9ae6-bbb0e6cbc4a7.png">

But this would mean, setting the `fontWeight` from theme might work but not when passed directly from the `style` unless `fontFamily` is also passed in.

Reply:
One more thing, if we do want to add `fontStyle` and `fontWeight` to `theme.button.style`, should we add `label` prefix to them so they became `labelFontStyl` and `labelFontWeight` or maybe there's no need? 🤔

Reply:
I agree to add `label` prefix.

> But this would mean, setting the `fontWeight` from theme might work but not when passed directly from the `style` unless `fontFamily` is also passed in.

I think this might confuse the user and also there can be a case where the user needs to use an array of style for the `labelStyles`.

Reply:
Personally, I think the theme should be flattened, so we will not support:
```
{
  text: { style: [{fontSize: 16}, {color: 'lightgreen'}]}
}
```

Reply:
Any particular reason for that?

Reply:
Keeping it flat when possible would make it easier to merge.

Remember that we receive partial theme in the provider and then it'll be merged with the default theme?

I guess the `mergeTheme` hasn't been updated since we support passing components style to the theme.

But right now, the theme could accept something like:
```
let customTheme = {
  style: {
    text: {
      style: { fontSize: 7 },
    },
    button: {
      labelStyle: [{ fontSize: 20 }, { fontWeight: '500' }],
    },
  },
};
```
While the `theme.style` works as additional style that will be passed to the individual components as props. We have not handled the merging with the default theme yet.

Let's say we have this for default theme.
```
let defaultTheme = {
  style: {
    button: {
      labelStyle: { fontSize: 14 },
    },
  },
};
```

This means now we need to handle merging style where some of them could be array, some of them are plain object.

While it's not necessarily difficult, I would argue we don't need this complexity in the first place.

Hmm, let's reverse the question.

Do we have a use case where the user would want to use an array for a style prop in the theme?

Reply:
Interesting. Makes sense!

I think I'll do just for button style in this PR and will put another in different PR.

> Do we have a use case where the user would want to use an array for a style prop in the theme?

Now you asking it makes me realize that there is no way a user would put an array style based some condition at theme initialization!

Reply:
I agree to make all the theme styles drop `StyleProp` but I think letting the user change the `fontWeight` through the `themeStyle` might not be a good idea.

Our `fontWeight` type is not the same type from `TextStyle`. It might be not a problem on the compilation as we can cast `FontWeight` type on it. But on the runtime, it can be a problem when the user define any font-weight that is not part of `FontWeight`,

Reply:
> Our `fontWeight` type is not the same type from `TextStyle`. It might be not a problem on the compilation as we can cast `FontWeight` type on it. But on the runtime, it can be a problem when the user define any font-weight that is not part of `FontWeight`,

I think in this case, we can omit the `fontWeight` from `TextStyle` and provide our own type for it. This might need a note somewhere in the theme docs though.

Reply:
Interesting! But I think it's better to go with `labelFontWeight` for the consistency as we can't change the font-weight through the Text's `TextStyle` to begin with. So it makes sense if we also can't change the font-weight through the theme's `TextStyle`, isn't it?

What do you think?

Reply:
I'm fine either way as long as the user could specify a font weight to use in the theme.